### PR TITLE
fix(oas3): change Server Variables parent

### PR DESCRIPTION
**Pull request by marcofriso**

### CODE REVIEW

**Comment by opichals** [comment]
Please add tests for the setters/getters to exemplify the use-cases.

File: packages/api-elements/test/api-description-test.js [hunk 1]
```diff
@@ -943,6 +1007,18 @@ describe('API description namespace', () => {
       });
     });
 
+    it('should have hosts', () => {
+      expect(transition.hosts.first.attributes.get('href').toValue()).to.deep.equal('https://first-server.com');
```
Comment (kylef) [inline packages/api-elements/test/api-description-test.js:1011]:
```suggestion
      expect(transition.hosts.first.href.toValue()).to.deep.equal('https://first-server.com');
```

Would the `href` accessor on resource work here?

File: packages/api-elements/test/api-description-test.js [hunk 2]
```diff
@@ -943,6 +1007,18 @@ describe('API description namespace', () => {
...
+    });
+
+    it('should set hosts', () => {
+      const host = new namespace.elements.Resource();
+      host.href = 'https://second-server.com';
+      transition.hosts = [host];
+
+      expect(transition.hosts.first.attributes.get('href').toValue()).to.deep.equal('https://second-server.com');
```
Comment (kylef) [inline packages/api-elements/test/api-description-test.js:1019]:
```suggestion
      expect(transition.hosts.first.href.toValue()).to.deep.equal('https://second-server.com');
```

File: packages/api-elements/test/api-description-test.js (hunk 1 above)
Comment (marcofriso) [reply packages/api-elements/test/api-description-test.js]: changed

File: packages/api-elements/test/api-description-test.js (hunk 2 above)
Comment (marcofriso) [reply packages/api-elements/test/api-description-test.js]: changed

**Comment by marcofriso** [comment]
@kylef is there something that prevent us to merge this?

**Pull request merged**